### PR TITLE
.github/workflows: cd workflow

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  push:
+    branches:
+      - master
+jobs:
+  test:
+    uses: ./.github/workflows/ci-go.yaml
+    secrets: inherit
+  build:
+    needs: test
+    runs-on: ubuntu-latest
```
Comment (dorav) [inline .github/workflows/cd.yaml:20]: building only for ubuntu?

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3.0.2
+        with:
+          ref: ${{ github.event.inputs.commit || 'master' }}
+      - name: Setup Go Environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.21'
```
Comment (dorav) [inline .github/workflows/cd.yaml:29]: go-file-version exists and takes the version from go.mod :)

File: .github/workflows/cd.yaml (hunk 1 above)
Comment (rotemtam) [reply .github/workflows/cd.yaml:20]: for now it will suffice, but sure we can add it soon

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (rotemtam) [inline .github/workflows/cd.yaml:29]:
```suggestion
          go-version-file: go.mod
```

File: .github/workflows/cd.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  build:
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3.0.2
+        with:
+          ref: ${{ github.event.inputs.commit || 'master' }}
+      - name: Setup Go Environment
+        uses: actions/setup-go@v2
```
Comment (rotemtam) [inline .github/workflows/cd.yaml:27]:
```suggestion
        uses: actions/setup-go@v4
```

**Pull request merged**